bundles/org.openhab.binding.sncf/src/main/java/org/openhab/binding/sncf/internal/handler/StationHandler.java

```diff
@@ -0,0 +1,223 @@
+                    getThing().getChannels().stream().map(Channel::getUID)
+                            .filter(channelUID -> isLinked(channelUID) && GROUP_ARRIVAL.equals(channelUID.getGroupId()))
+                            .forEach(channelUID -> {
+                                State state = getValue(channelUID.getIdWithoutGroup(), arrival, GROUP_ARRIVAL);
+                                updateState(channelUID, state);
+                            });
+                    scheduleRefresh(fromDTO(arrival.stopDateTime.arrivalDateTime));
+                }, () -> {
+                    logger.debug("No {} available", GROUP_ARRIVAL);
+                    scheduleRefresh(ZonedDateTime.now().plusMinutes(5));
+                });
+                updateStatus(ThingStatus.ONLINE);
+            } catch (SncfException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+                freeRefreshJob();
+            }
+        }
+    }
+
+    private State getValue(String channelId, Passage passage, String groupDeparture) {
+        switch (channelId) {
+            case DIRECTION:
+                return new StringType(passage.route.direction.name);
+            case CODE:
+                return new StringType(passage.displayInformations.code);
+            case COMMERCIAL_MODE:
+                return new StringType(passage.displayInformations.commercialMode);
+            case NAME:
+                return new StringType(passage.displayInformations.name);
+            case NETWORK:
+                return new StringType(passage.displayInformations.network);
+            case TIMESTAMP:
+                return new DateTimeType(
+                        fromDTO(groupDeparture.equals(GROUP_ARRIVAL) ? passage.stopDateTime.arrivalDateTime
+                                : passage.stopDateTime.departureDateTime));
+        }
+        return UnDefType.NULL;
+    }
+
+    private ZonedDateTime fromDTO(String dateTime) {
+        return ZonedDateTime.parse(dateTime + zoneOffset, NAVITIA_DATE_FORMAT);
+    }
+
+    private void freeRefreshJob() {
+        ScheduledFuture<?> job = refreshJob;
+        if (job != null) {
+            job.cancel(true);
+            this.refreshJob = null;
+        }
+    }
+
+    @Override
+    public void dispose() {
+        freeRefreshJob();
+        super.dispose();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            queryApiAndUpdateChannels();
+        }
+    }
+
+    private @Nullable SncfBridgeHandler getBridgeHandler() {
+        Bridge bridge = getBridge();
+        if (bridge != null) {
+            BridgeHandler handler = bridge.getHandler();
+            if (handler != null) {
+                if (handler.getThing().getStatus() == ThingStatus.ONLINE) {
```

Comment:
If bridge status is not ONLINE, it would be better to set BRIDGE_OFFLINE rather than BRIDGE_UNINITIALIZED

Reply:
Correct.

Reply:
You misunderstood my comment: this was appropriate to use BRIDGE_UNINITIALIZED when there is no bridge or no bridge handler. 
Use BRIDGE_OFFLINE when there is a bridge handler but the bridge is OFFLINE.

Reply:
Ok, corrected.